extensions/pkg/terraformer/config.go

```diff
@@ -245,17 +246,29 @@ func (t *terraformer) CleanupConfiguration(ctx context.Context) error {
 	t.logger.Info("Cleaning up all terraformer configuration")
 
 	t.logger.V(1).Info("Deleting Terraform variables Secret", "name", t.variablesName)
-	if err := t.client.Delete(ctx, &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.variablesName}}); err != nil && !apierrors.IsNotFound(err) {
+	varObject := &corev1.Secret{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.variablesName}}
+	if err := extensionscontroller.DeleteAllFinalizers(ctx, t.client, varObject); err != nil {
+		return err
+	}
+	if err := t.client.Delete(ctx, varObject); err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
 
 	t.logger.V(1).Info("Deleting Terraform configuration ConfigMap", "name", t.configName)
-	if err := t.client.Delete(ctx, &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.configName}}); err != nil && !apierrors.IsNotFound(err) {
+	configObject := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.configName}}
+	if err := extensionscontroller.DeleteAllFinalizers(ctx, t.client, configObject); err != nil {
+		return err
+	}
+	if err := t.client.Delete(ctx, configObject); err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
 
+	stateObject := &corev1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Namespace: t.namespace, Name: t.stateName}}
```

Comment:
Nit: Can you exchange L266 with L267 to keep it similar with the above invocations?